### DISCUSSION TITLE
How to append a value retrieved from the API

### DISCUSSION THREAD

**Discussion by Ryomasao** [react-hook-form/react-hook-form#7214] [Q&A]
When append is executed asynchronously, the data passed to append is not set.
Is this the expected behavior?

> You can not call actions one after another. Actions need to be triggered per render.

Do the following cases also apply to the above rules?

```tsx
  const handleDelayAppend = () => {
    setTimeout(() => {
      // Initial value is not set.
      append({
        name: "foo",
        age: 12
      });
    }, 100);
  };
```

**To Reproduce**

Go to https://codesandbox.io/s/asycn-append-ge1vk?file=/src/App.tsx
click `append delay` button

**ScreenShot**

https://user-images.githubusercontent.com/8813191/144738540-26b1f9f0-0675-47b8-bda7-0d3c7a6624ec.mp4

**Comment by bluebill1049** [comment]
issue created: https://github.com/react-hook-form/react-hook-form/issues/7217
